csi/src/server.rs

```diff
@@ -191,16 +213,37 @@ async fn main() -> Result<(), String> {
         }
     }
 
-    let mut uds_sock = UnixListener::bind(csi_socket).unwrap();
-    info!("CSI plugin bound to {}", csi_socket);
-
-    let uds = Server::builder()
-        .add_service(NodeServer::new(Node {
-            node_name: node_name.into(),
-            filesystems: probe_filesystems(),
-        }))
-        .add_service(IdentityServer::new(Identity {}))
-        .serve_with_incoming(uds_sock.incoming().map_ok(UnixStream));
-    let _ = uds.await;
+    let grpc_endpoint = format!("{}:{}", ipaddr, default_port());
+    let fut_csi = CSIServer::run(csi_socket, node_name);
+    let fut_grpc =
+        MayastorNodePluginGrpcServer::run(grpc_endpoint.parse().unwrap());
+    let futs = join(fut_csi, fut_grpc);
+    let mut rt = Runtime::new().expect("failed to obtain a new Runtime object");
+    let _ = rt.block_on(futs);
     Ok(())
 }
+
+struct CSIServer {}
+
+impl CSIServer {
+    pub async fn run(csi_socket: &str, node_name: &str) -> Result<(), ()> {
+        let mut uds_sock = UnixListener::bind(csi_socket).unwrap();
+        info!("CSI plugin bound to {}", csi_socket);
+
+        let uds = Server::builder()
+            .add_service(NodeServer::new(Node {
+                node_name: node_name.into(),
+                filesystems: probe_filesystems(),
+            }))
+            .add_service(IdentityServer::new(Identity {}))
+            .serve_with_incoming(uds_sock.incoming().map_ok(UnixStream));
+
+        match uds.await {
+            Ok(_) => Ok(()),
+            Err(e) => {
+                println!("CSI node server failed with error: {}", e);
```

Comment:
probably should be using error! here

Reply:
Done. https://github.com/openebs/Mayastor/pull/462/commits/7c714899d911a2f659d5a7861968ba08b1aa354b